google/cloud/spanner/integration_tests/backup_integration_test.cc

```diff
@@ -36,116 +37,115 @@ namespace spanner {
 inline namespace SPANNER_CLIENT_NS {
 namespace {
 
-class BackupTest : public testing::Test {
+std::string const& ProjectId() {
+  static std::string project_id =
+      google::cloud::internal::GetEnv("GOOGLE_CLOUD_PROJECT").value_or("");
+  return project_id;
+}
+
+bool RunSlowBackupTests() {
+  static bool run_slow_backup_tests =
+      google::cloud::internal::GetEnv(
+          "GOOGLE_CLOUD_CPP_SPANNER_SLOW_INTEGRATION_TESTS")
+          .value_or("")
+          .find("backup") != std::string::npos;
+  return run_slow_backup_tests;
+}
+
+bool Emulator() {
+  static bool emulator =
+      google::cloud::internal::GetEnv("SPANNER_EMULATOR_HOST").has_value();
+  return emulator;
+}
+
+class CleanupStaleInstances : public ::testing::Environment {
+ public:
+  void SetUp() override {
+    std::regex instance_name_regex(
+        R"(projects/.+/instances/)"
+        R"((temporary-instance-(\d{4}-\d{2}-\d{2})-.+))");
+
+    // Make sure we're using a correct regex.
+    EXPECT_EQ(2, instance_name_regex.mark_count());
+    auto generator = google::cloud::internal::MakeDefaultPRNG();
+    Instance in(ProjectId(), spanner_testing::RandomInstanceName(generator));
+    auto fq_instance_name = in.FullName();
+    std::smatch m;
+    EXPECT_TRUE(std::regex_match(fq_instance_name, m, instance_name_regex));
+    EXPECT_EQ(3, m.size());
+
+    if (RunSlowBackupTests()) {
```

Comment:
You can probably perform this cleanup regardless of whether slow tests are enabled or not, up to you.

Reply:
I'll leave it as is for now, but will reassess when I return here soon for the PITR/CMEK tests.